lib/shim/datastore-shim.js

```diff
@@ -693,6 +694,7 @@ function _recordQuery(suffix, nodule, properties, querySpec) {
       promise: 'promise' in queryDesc ? queryDesc.promise : null,
       internal: 'internal' in queryDesc ? queryDesc.internal : true,
       opaque: 'opaque' in queryDesc ? queryDesc.opaque : false,
+      inContext: 'inContext' in queryDesc ? queryDesc.inContext : null,
```

Comment:
Neat, didn't know segment descriptors could have this attribute.

Reply:
yea it's used in message-shim. I didn't add support for this in all cases in here as this is the only use case atm